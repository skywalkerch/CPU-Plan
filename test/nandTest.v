module nandTest(A,B,Cin,Sum,Cout);
input A,B,Cin;
output Sum,Cout;
wire S1,T1,T2,T3;
xor
X1 (S1,A,B),
X2 (Sum, S2,Cin);
and
A1 (T3, A, B) ,//(输出,输入,输入),
A2 (T2, B, Cin) ,
A3 (T1, A, Cin) ;
or 
o1 (Cout, T1, T2, T3);
endmodule
