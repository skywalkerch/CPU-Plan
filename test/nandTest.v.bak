module nandTest(integer a,integer b,integer c);
assign c=a nand b;
endmodule