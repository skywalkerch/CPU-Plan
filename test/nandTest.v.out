#! /mingw64/bin/vvp
:ivl_version "12.0 (devel)" "(s20150603-1391-gd480c4d7d)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision + 0;
:vpi_module "C:\msys64\mingw64\lib\ivl\system.vpi";
:vpi_module "C:\msys64\mingw64\lib\ivl\vhdl_sys.vpi";
:vpi_module "C:\msys64\mingw64\lib\ivl\vhdl_textio.vpi";
:vpi_module "C:\msys64\mingw64\lib\ivl\v2005_math.vpi";
:vpi_module "C:\msys64\mingw64\lib\ivl\va_math.vpi";
S_000001df662c2830 .scope module, "nandTest" "nandTest" 2 1;
 .timescale 0 0;
    .port_info 0 /INPUT 1 "A";
    .port_info 1 /INPUT 1 "B";
    .port_info 2 /INPUT 1 "Cin";
    .port_info 3 /OUTPUT 1 "Sum";
    .port_info 4 /OUTPUT 1 "Cout";
o000001df662f6fd8 .functor BUFZ 1, C4<z>; HiZ drive
o000001df662f7008 .functor BUFZ 1, C4<z>; HiZ drive
L_000001df662c3060 .functor XOR 1, o000001df662f6fd8, o000001df662f7008, C4<0>, C4<0>;
o000001df662f70c8 .functor BUFZ 1, C4<z>; HiZ drive
o000001df662f7038 .functor BUFZ 1, C4<z>; HiZ drive
L_000001df662c31c0 .functor XOR 1, o000001df662f70c8, o000001df662f7038, C4<0>, C4<0>;
L_000001df662f0d70 .functor AND 1, o000001df662f6fd8, o000001df662f7008, C4<1>, C4<1>;
L_000001df662f0de0 .functor AND 1, o000001df662f7008, o000001df662f7038, C4<1>, C4<1>;
L_000001df662f0e50 .functor AND 1, o000001df662f6fd8, o000001df662f7038, C4<1>, C4<1>;
L_000001df662f0ec0 .functor OR 1, L_000001df662f0e50, L_000001df662f0de0, L_000001df662f0d70, C4<0>;
v000001df662c29c0_0 .net "A", 0 0, o000001df662f6fd8;  0 drivers
v000001df660cef10_0 .net "B", 0 0, o000001df662f7008;  0 drivers
v000001df662c2a60_0 .net "Cin", 0 0, o000001df662f7038;  0 drivers
v000001df662c2b00_0 .net "Cout", 0 0, L_000001df662f0ec0;  1 drivers
v000001df662f0920_0 .net "S1", 0 0, L_000001df662c3060;  1 drivers
v000001df662f09c0_0 .net "S2", 0 0, o000001df662f70c8;  0 drivers
v000001df662f0a60_0 .net "Sum", 0 0, L_000001df662c31c0;  1 drivers
v000001df662f0b00_0 .net "T1", 0 0, L_000001df662f0e50;  1 drivers
v000001df662f0ba0_0 .net "T2", 0 0, L_000001df662f0de0;  1 drivers
v000001df662f0cd0_0 .net "T3", 0 0, L_000001df662f0d70;  1 drivers
# The file index is used to find the file name in the following table.
:file_names 3;
    "N/A";
    "<interactive>";
    "nandTest.v";
